commit e9cd2aaa692af94d656da2b97bd8b54f2d5c8627
Author: Cody Peter Mello <cody.mello@joyent.com>
Date:   2019-07-09T21:31:01+00:00 (3 months ago)
    
    OS-5474 mdata-get does not properly report tags that include periods
    OS-6331 Make it easier to find MetadataAgent instances in core dumps

diff --git a/src/vm/lib/metadata/agent.js b/src/vm/lib/metadata/agent.js
index ea5a1a5c..10999050 100644
--- a/src/vm/lib/metadata/agent.js
+++ b/src/vm/lib/metadata/agent.js
@@ -20,7 +20,7 @@
  *
  * CDDL HEADER END
  *
- * Copyright (c) 2019, Joyent, Inc.
+ * Copyright 2019 Joyent, Inc.
  *
  *
  * # OVERVIEW
@@ -158,6 +158,10 @@ var PERIODIC_CONNECTION_RETRY = 60 * 1000; // every minute
 
 function noop() {}
 
+function hasKey(obj, key) {
+    return (Object.prototype.hasOwnProperty.call(obj, key));
+}
+
 function closeZoneConnection(zoneConn) {
     assert.object(zoneConn, 'zoneConn');
 
@@ -199,13 +203,13 @@ function elapsedTimer(timer) {
 }
 
 
-var MetadataAgent = module.exports = function (options) {
+function MetadataAgent(options) {
     this.log = options.log;
     this.zlog = {};
     this.zonesDebug = {};
     this.zoneConnections = {};
     this.zoneKvmReconnTimers = {};
-};
+}
 
 /*
  * This function exists to add debug information to the zonesDebug object. That
@@ -1129,6 +1133,13 @@ MetadataAgent.prototype.makeMetadataHandler = function (zone, socket) {
                     returnit(null, vmobj.internal_metadata['operator-script']);
                 } else if (want === 'volumes') {
                     returnit(null, vmobj.internal_metadata['sdc:volumes']);
+                } else if (want.slice(0, 5) === 'tags.') {
+                    want = want.slice(5);
+                    if (hasKey(vmobj.tags, want)) {
+                        val = vmobj.tags[want];
+                    }
+
+                    returnit(null, val);
                 } else {
                     val = VM.flatten(vmobj, want);
                     returnit(null, val);
@@ -1439,3 +1450,5 @@ MetadataAgent.prototype.makeMetadataHandler = function (zone, socket) {
         }
     };
 };
+
+module.exports = MetadataAgent;
diff --git a/src/vm/node_modules/qmp.js b/src/vm/node_modules/qmp.js
index ff9443e1..871a06c1 100644
--- a/src/vm/node_modules/qmp.js
+++ b/src/vm/node_modules/qmp.js
@@ -20,7 +20,7 @@
  *
  * CDDL HEADER END
  *
- * Copyright (c) 2014 Joyent Inc., All rights reserved.
+ * Copyright 2019 Joyent, Inc.
  *
  */
 
@@ -32,7 +32,7 @@ var bunyan = require('/usr/node/node_modules/bunyan');
 var events = require('events');
 var net = require('net');
 
-var Qmp = function (logger)
+function Qmp(logger)
 {
     var id;
     var ids = [];
@@ -75,7 +75,7 @@ var Qmp = function (logger)
             qmp.log.debug('Got "send", but not connected.');
         }
     });
-};
+}
 
 Qmp.prototype.send = function (msg)
 {
